File: codes1/branWh.py
import math
import numpy as np
from travWh import travWh
def branWh():
    L = math.pi
    N = 2*2*16
    h = float(L)/N

    x = np.arange(h/2.,L,h)
    cstar = math.sqrt(math.tanh(1))
    c = 0.99*cstar
    vini = -0.015-0.1*np.cos(x)
    tolerance =  1e-12
    
    u = travWh(c,vini,N,tolerance)

    for i in range(1,40):
        c = c - 0.0025
        u = travWh(c,u,N,tolerance)
    return u
    
def main():
    wave = branWh()
    print wave
    
if __name__ == '__main__':
  main()

File: codes1/Sqrt version/branWh.m
%
% This program computes a branch of traveling waves of 
%
% -cu + + u^2 + K_h_0 u = 0   on 0 < x < pi
%

%clear;

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%
%%
%
%  Basic parameters
%
%%
%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
L = pi;
N = 2*2*16; 
h = L/N;


x = (h/2:h:L-h/2)';   % cosine grid [h/2, 3h/2, 5h/2, .... , L-3h/2, L-h/2]


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%
%%
%
%  Initial guess
% 
%%
%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


cstar = sqrt(tanh(1));
c = 0.99*cstar;
vini = -0.015-0.1*cos(x);


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%
%%
%
%  Call functions
% 
%%
%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


tolerance =  1e-12;
u = travWh(c,vini,N,tolerance);


for ind=2:40;
   c=c-0.0025;
   u = travWh(c,u,N,tolerance);
end;

plot(x,u);
